clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -136,27 +369,160 @@ public ConfigDef withClientSaslSupport() {
      * the appropriate type (int, string, etc)
      */
     public Map<String, Object> parse(Map<?, ?> props) {
+        return parse(props, true);
+    }
+
+    /**
+     * Parse and validate configs against this configuration definition. The input is a map of configs. It is expected
+     * that the keys of the map are strings, but the values can either be strings or they may already be of the
+     * appropriate type (int, string, etc). This will work equally well with either java.util.Properties instances or a
+     * programmatically constructed map.
+     *
+     * @param props The configs to parse and validate
+     * @param checkRequired Whether to throw a config exception in case that we don't provide a value for a required config
+     * @return Parsed and validated configs. The key will be the config name and the value will be the value parsed into
+     * the appropriate type (int, string, etc)
+     */
+    public Map<String, Object> parse(Map<?, ?> props, boolean checkRequired) {
```

Comment:
This is used in the JDBC/HDFS connector. When we set check required to be false, if we cannot parse the value, a ConfigException is thrown and will be be caught in the validate method.  `ConfigDef.parse` with `checkRequired` false  will only be used in the recommender class. In this case, if we cannot parse the value, we should throw the exception instead of catching it.